src/messagequeue/adapters/eventhub.ts

```diff
@@ -23,8 +23,42 @@ export const create = (
     await producer.close();
   };
 
+  const sendMessagesBatch = async(messages: Message[], batchOptions: CreateBatchOptions) => {
+    try {
+      const producer: EventHubProducerClient = new EventHubProducerClient(
+        connectionString,
+        name
+      );
+
+      let batch = await producer.createBatch(batchOptions);
+      let batchesSent = 0
+
+      for (let i = 0; i < messages.length; i++ )
```

Comment:
higher order loop like `for {const message of messages}` would eliminate the need to track indices here and below.

You could perhaps get away from a loop entirely by doing a mapping with `messages.map(message => { ... }`

Reply:
I originally did it in a for loop because it was easier to handle the asynchronous operations from each time it makes or sends a batch.

I can rewrite it to use map, and then resolve the promise before continuing